protocol/messenger.go

```diff
@@ -2089,59 +2089,40 @@ func (m *Messenger) DeleteChat(chatID string) error {
 	return nil
 }
 
-func (m *Messenger) isNewContact(contact *Contact) bool {
-	previousContact, ok := m.allContacts[contact.ID]
-	return contact.IsAdded() && (!ok || !previousContact.IsAdded())
+func (m *Messenger) DeactivateChat(chatID string) (*MessengerResponse, error) {
+	m.mutex.Lock()
+	defer m.mutex.Unlock()
+	return m.deactivateChat(chatID)
 }
 
-func (m *Messenger) hasNicknameChanged(contact *Contact) bool {
-	previousContact, ok := m.allContacts[contact.ID]
+func (m *Messenger) deactivateChat(chatID string) (*MessengerResponse, error) {
+	var response MessengerResponse
+	chat, ok := m.allChats[chatID]
 	if !ok {
-		return false
+		return nil, ErrChatNotFound
 	}
-	return contact.LocalNickname != previousContact.LocalNickname
-}
 
-func (m *Messenger) removedContact(contact *Contact) bool {
-	previousContact, ok := m.allContacts[contact.ID]
-	if !ok {
-		return false
-	}
-	return previousContact.IsAdded() && !contact.IsAdded()
-}
+	clock, _ := chat.NextClockAndTimestamp(m.getTimesource())
+
+	err := m.persistence.DeactivateChat(chat, clock)
 
-func (m *Messenger) saveContact(contact *Contact) error {
-	name, identicon, err := generateAliasAndIdenticon(contact.ID)
 	if err != nil {
-		return err
+		return nil, err
 	}
 
-	contact.Identicon = identicon
-	contact.Alias = name
-
-	if m.isNewContact(contact) || m.hasNicknameChanged(contact) {
-		err := m.syncContact(context.Background(), contact)
+	if chat.Public() {
+		err := m.reregisterForPushNotifications()
```

Comment:
"reregister"? Shouldn't this be called "deregister"? Or am I missing what this actually does?

Reply:
it's re-registering effectively, as the fact that a contact has been removed means that we need to re-register with the PN server as our options have changed, I can add a comment